### PR TITLE
Add script that generates release assets

### PR DESCRIPTION
**Pull request by fbs**
Bit hacky but atleast we'll have consistent tars. Future version could
get the binary for us.

### CODE REVIEW

File: scripts/create-assets.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+info "Creating man archive"
+mkdir -p "$TMP/share/man/man8"
+cp man/man8/*.8 "$TMP/share//man/man8/"
+gzip "$TMP/share/man/man8/"*
+tar --xz -cf "$OUT/man.tar.xz" -C "$TMP/share" man
+
+info "Creating bundle"
+ln bpftrace "$TMP/bin/bpftrace"
+tar --xz -cf "$OUT/bundle.tar.xz" -C "$TMP" bin share
```
Comment (fbs) [inline scripts/create-assets.sh:46]:
```suggestion
tar --xz -cf "$OUT/binary+tools+man.tar.xz" -C "$TMP" bin share
```

not sure..

File: scripts/create-assets.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+[[ -f bpftrace ]] || err "bpftrace binary not found, download the build artifact"
+
+OUT=$(mktemp -d) || err "Failed to create temp dir"
+TMP="${OUT}/tmp"
+
+echo "Using '$OUT'"
+
+
+set -e
+
```
Comment (fbs) [inline scripts/create-assets.sh:23]: considering `xz`ing the binary as well, so we have bpftrace.xz as artifact.

File: scripts/create-assets.sh [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  exit 1
+}
+
+function info() {
+  echo "$(tput bold)$*$(tput sgr0)"
+}
+
+[[ -d tools ]] || err "'tools' directory not found, run script from bpftrace root dir"
+[[ -d man ]] || err "'man' directory not found, run script from bpftrace root dir"
+[[ -f bpftrace ]] || err "bpftrace binary not found, download the build artifact"
```
Comment (danobi) [inline scripts/create-assets.sh:14]: Might be better to figure out repo root from script path (`$0`) so script can be run from another dir.

File: docs/release_process.md [hunk 4]
```diff
@@ -42,9 +42,11 @@ You must do these things to formally release a version:
    as "commits"), then "Draft a new release". The tag version and release title
    should be the same and in `vX.Y.Z` format. The tag description should
    be the same as what you added to `CHANGELOG.md`.
-1. Attach a tar of the tools to the release.
 1. Once the release tag pipeline has finished, extract the bpftrace binary from
    the release build on and attach it to the release.
   -  `docker run -v
 $(pwd):/output quay.io/iovisor/bpftrace:vXX.YY.ZZ /bin/bash -c "cp
 /usr/bin/bpftrace /output"`
+1. Run `scripts/create-assets.sh` from bpftrace root dir  and attach the
```
Comment (danobi) [inline docs/release_process.md:50]:
```suggestion
1. Run `scripts/create-assets.sh` from bpftrace root dir and attach the
```

File: scripts/create-assets.sh (hunk 1 above)
Comment (danobi) [reply scripts/create-assets.sh:46]: I think verbose is better

File: scripts/create-assets.sh (hunk 2 above)
Comment (danobi) [reply scripts/create-assets.sh:23]: If we're going to compress might as well go w/ best algorithm (zstd maybe?)

Comment (fbs) [reply scripts/create-assets.sh:23]: guess we could just do both. Zstd is not that common yet, will probably confuse people. If we XZ it we still have a nice saving

File: scripts/create-assets.sh (hunk 3 above)
Comment (fbs) [reply scripts/create-assets.sh:14]: I considered it but was hoping to keep this script minimal in options. I only need this a few times a year, copying the binary/symlinking the binary isn't that much work

**Comment by fbs** [comment]
Added zstd version of the bundle and binary for the coolkids that prefer zstd. 

Size diff is minimal

```
 13M	binary_tools_man-bundle.tar.xz
 14M	binary_tools_man-bundle.tar.zst
 46M	bpftrace
 13M	bpftrace.xz
 14M	bpftrace.zst
 40K	man.tar.xz
 24K	tools-with-help.tar.xz
 12K	tools.tar.xz
```

```
26736	binary_tools_man-bundle.tar.xz
28800	binary_tools_man-bundle.tar.zst
93880	bpftrace
26720	bpftrace.xz
28800	bpftrace.zst
```

**Comment by danobi** [comment]
Ok let's stick to xz then

**Review by danobi — APPROVED** [review]

**Pull request merged**